Formula/texlive.rb

```diff
@@ -0,0 +1,523 @@
+    rm bin/"latexdiff" # provided by latexdiff formula
+    rm bin/"latexdiff-vc" # provided by latexdiff formula
+    rm bin/"latexrevise" # provided by latexdiff formula
+
+    # Wrap some Perl scripts in an env script so that they can find dependencies
+    env_script_files = %w[
+      crossrefware/bbl2bib.pl
+      crossrefware/bibdoiadd.pl
+      crossrefware/bibmradd.pl
+      crossrefware/biburl2doi.pl
+      crossrefware/bibzbladd.pl
+      crossrefware/ltx2crossrefxml.pl
+      ctan-o-mat/ctan-o-mat.pl
+      ctanify/ctanify
+      ctanupload/ctanupload.pl
+      exceltex/exceltex
+      latex-git-log/latex-git-log
+      pax/pdfannotextractor.pl
+      ptex-fontmaps/kanji-fontmap-creator.pl
+      purifyeps/purifyeps
+      svn-multi/svn-multi.pl
+      texdoctk/texdoctk.pl
+      ulqda/ulqda.pl
+    ]
+
+    env_script_files.each do |perl_script|
+      bin_name = File.basename(perl_script, ".pl")
+      rm bin/bin_name
+      (bin/bin_name).write_env_script(share/"texmf-dist/scripts"/perl_script, PERL5LIB: ENV["PERL5LIB"])
+    end
+
+    # Wrap some Python scripts so they can find dependencies and fix depythontex.
+    python_path = libexec/Language::Python.site_packages("python3")
+    ENV.prepend_path "PYTHONPATH", python_path
+    rm bin/"pygmentex"
+    rm bin/"pythontex"
+    rm bin/"depythontex"
+    (bin/"pygmentex").write_env_script(share/"texmf-dist/scripts/pygmentex/pygmentex.py",
+      PYTHONPATH: ENV["PYTHONPATH"])
+    (bin/"pythontex").write_env_script(share/"texmf-dist/scripts/pythontex/pythontex3.py",
+      PYTHONPATH: ENV["PYTHONPATH"])
+    ln_sf share/"texmf-dist/scripts/pythontex/depythontex3.py", bin/"depythontex"
+
+    # Rewrite shebangs in some Python scripts so they use brewed Python.
+    python_shebang_rewrites = %w[
+      dviasm/dviasm.py
+      latex-make/figdepth.py
+      latex-make/gensubfig.py
+      latex-make/latexfilter.py
+      latex-make/svg2dev.py
+      latex-make/svgdepth.py
+      latex-papersize/latex-papersize.py
+      lilyglyphs/lilyglyphs_common.py
+      lilyglyphs/lily-glyph-commands.py
+      lilyglyphs/lily-image-commands.py
+      lilyglyphs/lily-rebuild-pdfs.py
+      pdfbook2/pdfbook2
+      pygmentex/pygmentex.py
+      pythontex/depythontex3.py
+      pythontex/pythontex3.py
+      pythontex/pythontex_install.py
+      spix/spix.py
+      texliveonfly/texliveonfly.py
+      webquiz/webquiz
+      webquiz/webquiz.py
+      webquiz/webquiz_makequiz.py
+      webquiz/webquiz_util.py
+    ]
+
+    python_shebang_rewrites.each do |python_script|
+      rewrite_shebang detected_python_shebang, share/"texmf-dist/scripts"/python_script
+    end
+
+    # Delete ebong because it requires Python 2
+    rm bin/"ebong"
+
+    # Initialize texlive environment
+    ENV.prepend_path "PATH", bin
+    system "fmtutil-sys", "--all"
+    system "mtxrun", "--generate"
+    system "mktexlsr"
+  end
+
+  test do
+    assert_match "Usage", shell_output("#{bin}/tex --help")
```

Comment:
A simple example of LaTeX document that you can test some packages installed:

```tex
\documentclass[12pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{lipsum}

\title{\LaTeX\ test}
\author{\TeX\ Team}
\date{September 2021}

\begin{document}

\maketitle

\section*{An equation with amsmath}
\begin{equation} \label{eu_eqn}
e^{\pi i} + 1 = 0
\end{equation}
The beautiful equation \ref{eu_eqn} is known as the Euler equation.

\section*{Lorem Ipsum}
\lipsum[3]

\lipsum[5]

\end{document}
```

You can compile with `tex`and `pdftex` to generate ps, pdf and dvi outputs, for example.

Reply:
You want `pdflatex`, I think, and just test for the existence of a `pdf` file afterwards.

Reply:
A few `\usepackage` statements would be good to include in the test to make sure that they're there.